File: UVa Online Judge/100-999/575/skewBinary_575.py


while(1):
	n = input()

	if(n == '0'):
		break

	l = len(n)
	total = 0
	nmax = 2**31 - 1

	for i in range(l):
		a = int(n[i]) * (2**l - 1)
		l-=1
		total+=a

		if(total > nmax):
			total = nmax
			break


	print(str(total))

File: UVa Online Judge/100-999/575/SkewBinary_575.cpp
#include <iostream>
#include <string>
#include <math.h>
using namespace std;

int main(){

	string n;
	int l, ll, total, a;
	int nmax = pow(2.0, 31) - 1;

	while(cin>>n && n.compare("0")!=0){
		l = n.length();
		ll = l;
		total = 0;

		for(int i=0; i<l; i++){
			a = n[i] - '0';
			a *= (pow(2.0, ll) - 1);
			ll--;
			total+=a;

			if(total > nmax){
				total = nmax;
				break;
			}
		}

		cout<<total<<'\n';

	}       

	return 0;
}
                    
